### DISCUSSION TITLE
Made a wrapper API around weather gov

### DISCUSSION THREAD

**Discussion by lemonprogis** [weather-gov/api#708] [Show and tell]
Hello! 

Wanted to share this everyone. I ended up merging two datasets that I always tend to get. I wanted current observations AND forecast data. So I built this to bring those two things together. 

![API Breakdown](https://github.com/lemonprogis/weathermate-backend/raw/main/images/diagram.png  "Overview")

1. Combine the calls to forecast and api weather.gov endpoints.
2. Get the forecasts for the gridpoint returned. Both daily and hourly.
3. Translate all of this into an easy to consume response. 

[Weathermateplus API](https://weathermateplus.com/api-documentation/)

Thanks for the hard work you guys put into this to enable folks like me to create.

Cheers!

**Comment by changliao1025** [comment]
Nice work! Can you also provide some (Python) examples for different applications?

**Reply by lemonprogis** [reply]
Thank you! Yes, I can definitely add some examples.

**Reply by lemonprogis** [reply]
@changliao1025 here is a sample of both endpoints using Python's requests library.

```
import json
import requests


def get_weather_data_by_location(lat, lng):
    url = f'https://weathermateplus.com/api'
    params = {
        'lat': lat,
        'lng': lng
    }
    response = requests.get(url, params=params)
    return response.json()


def get_weather_data_search(addy):
    url = f'https://weathermateplus.com/api/location'
    params = {
        'address': addy
    }
    response = requests.get(url, params=params)
    return response.json()


def pretty_print(j):
    print(json.dumps(j, indent=4))


if __name__ == '__main__':
    latitude = 46.4884768
    longitude = -120.18902

    search_term = 'Richland, WA'

    d1 = get_weather_data_by_location(latitude, longitude)
    d2 = get_weather_data_search(search_term)

    pretty_print(d1)
    pretty_print(d2)
```

Responses are the same, but you can get data from lat/lng pairs or from providing an address. Hope this helps!

**Reply by changliao1025** [reply]
Excellent! I will give it a try and let you know if I have any questions. I also found another Python package https://github.com/meteostat/meteostat-python, that has similar features.

**Reply by lemonprogis** [reply]
That's a good package. Also, if you want to run mine, I made a public docker image that you can deploy anywhere. 

https://hub.docker.com/r/lemonpro/weathermate
